src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs

```diff
@@ -264,13 +260,28 @@ internal IEnumerable<string> CreateArgument()
                 {
                     allArgs.Add("--testAdapterPath:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(this.VSTestTraceDataCollectorDirectoryPath));
                 }
+                else
+                {
+                    if (isCollectCodeCoverageEnabled)
+                    {
+                        // Not showing message in runsettings scenario, because we are not sure that code coverage is enabled.
+                        // User might be using older Microsoft.NET.Test.Sdk which don't have CodeCoverage infra.
+                        Console.WriteLine(Resources.UpdateTestSdkForCollectingCodeCoverage);
+                    }
+                }
             }
 
-            if (!string.IsNullOrEmpty(this.VSTestBlame))
+            if (this.VSTestCLIRunSettings != null && this.VSTestCLIRunSettings.Length > 0)
             {
-                allArgs.Add("--Blame");
+                allArgs.Add("--");
+                foreach (var arg in this.VSTestCLIRunSettings)
+                {
+                    allArgs.Add(ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(arg));
+                }
             }
 
+            // VSTestCLIRunSettings should always be last argument in allArgs. As vstest.console ignore options after "--"(CLIRunSettings option).
```

Comment:
Move this comment to place where we add "--".
Nit: VSTestCLIRunSettings should be last argument in allArgs as vstest.console ignore options after "--"

Reply:
Moved adding CLIRunSettings to separate method which calls after adding all the args.

Reply:
Updated the comment.